src/cci/cas_cci.c

```diff
@@ -712,14 +713,24 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT])
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      if (con_handle->id < 0)
+	{
+	  return CCI_ER_INVALID_ARGS;
+	}
+      MUTEX_LOCK (con_handle_table_mutex);
+      rv = hm_put_con_to_pool (con_handle->id);
+      MUTEX_UNLOCK (con_handle_table_mutex);
+      if (rv >= 0)
```

Comment:
Even if the id of con_handle cannot be put in the connection pool, it seems appropriate to handle disconnection.

Reply:
And mutex is also required to keep **con_handle->id** value.

Reply:
The rv value has nothing to do with con_handle->id.
rv is 0 if con_handle->id is added to the connection pool, -1 if not. ( Even if con_handle->id is -1, rv can be 0.)

Reply:
Would you explain why this commend is related with rv? I cannot understand.